bundles/binding/org.openhab.binding.fatekplc/src/main/java/org/openhab/binding/fatekplc/internal/FatekPLCBinding.java

```diff
@@ -0,0 +1,278 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.openhab.binding.fatekplc.FatekPLCBindingProvider;
+import org.openhab.binding.fatekplc.items.FatekPLCItem;
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.s4u.jfatek.io.FatekIOException;
+
+/**
+ * Binding implementation for Fatek PLC.
+ *
+ * @author Slawomir Jaranowski
+ * @since 1.9.0
+ */
+public class FatekPLCBinding extends AbstractActiveBinding<FatekPLCBindingProvider> implements ManagedService {
+
+	/**
+	 * Main logger.
+	 */
+	private static final Logger logger = LoggerFactory.getLogger(FatekPLCBinding.class);
+
+	/**
+	 * Globals config keys name
+	 */
+	private static final List<String> GLOBAL_CONFIG_NAMES = Arrays.asList("refresh", "service.pid");
+
+	/**
+	 * the refresh interval which is used to poll values from the Fatek PLC server (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 60000;
+
+	/**
+	 * Map of slave device.
+	 */
+	private Map<String, FatekPLCSlave> slaves = new HashMap<>();
+
+	/**
+	 * ItemRegistry service.
+	 */
+	private ItemRegistry itemRegistry;
+
+	public FatekPLCBinding() {
+	}
+
+	/**
+	 * Called by OSGI.
+	 *
+	 * @param itemRegistry
+	 */
+	public void setItemRegistry(ItemRegistry itemRegistry) {
+		this.itemRegistry = itemRegistry;
+	}
+
+	/**
+	 * Called by OSGI.
+	 *
+	 * @param itemRegistry
+	 */
+	public void unsetItemRegistry(ItemRegistry itemRegistry) {
+		this.itemRegistry = null;
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "Fatek PLC Refresh Service";
+	}
+
+	/**
+	 * Called by the SCR to deactivate the component when either the configuration is removed or mandatory references
+	 * are no longer satisfied or the component has simply been stopped.
+	 *
+	 * @param reason
+	 *            Reason code for the deactivation:<br>
+	 *            <ul>
+	 *            <li>0 Unspecified
+	 *            <li>1 The component was disabled
+	 *            <li>2 A reference became unsatisfied
+	 *            <li>3 A configuration was changed
+	 *            <li>4 A configuration was deleted
+	 *            <li>5 The component was disposed
+	 *            <li>6 The bundle was stopped
+	 *            </ul>
+	 */
+	public void deactivate(final int reason) {
+
+		setProperlyConfigured(false);
+		logger.info("deactivate executed reason={}", reason);
+	}
+
+	@Override
+	public void updated(Dictionary<String, ?> properties) throws ConfigurationException {
+
+		logger.debug("update start");
+		try {
+			setProperlyConfigured(false);
```

Comment:
Ok, I remove this.
At end of methods if no exceptions occurs `setProperlyConfigured(true);` is called.